### PR TITLE
[EVM] Add EVM type ID

### PR DESCRIPTION
**Pull request by sideninja**
Closes: #5232

This PR adds a new type ID for EVM which can be used in events. 

`evm.BlockExecuted` and `evm.TransactionExecuted`

### CODE REVIEW

**Review by turbolent — APPROVED** [review]
Nice!

File: fvm/evm/types/events.go [hunk 1]
```diff
@@ -26,6 +31,72 @@ type Event struct {
...
+
+func (l EVMLocation) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Type string
+	}{
+		Type: "EVMLocation",
+	})
+}
+
+func init() {
```
Comment (sideninja) [inline fvm/evm/types/events.go:75]: I don't like using init but not sure if there's a better way of doing this without exporting types.

Comment (janezpodhostnik) [reply fvm/evm/types/events.go:75]: Looks like these are usually in init... Not sure if there is another way.

File: fvm/fvm_test.go [hunk 2]
```diff
@@ -3013,4 +3013,65 @@ func TestEVM(t *testing.T) {
 			))
 		}),
 	)
+
+	t.Run("deploy contract code", newVMTest().
+		withBootstrapProcedureOptions(fvm.WithSetupEVMEnabled(true)).
+		// we keep this dissabled during bootstrap and later overwrite in the test for test transaction
```
Comment (janezpodhostnik) [inline fvm/fvm_test.go:3019]: you don't need to do this anymore right.

File: fvm/evm/types/events.go [hunk 3]
```diff
@@ -26,6 +31,72 @@ type Event struct {
...
+	common.RegisterTypeIDDecoder(
+		evmLocationPrefix,
+		func(_ common.MemoryGauge, typeID string) (common.Location, string, error) {
```
Comment (janezpodhostnik) [inline fvm/evm/types/events.go:78]: Do we need a test for this part specifically?

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/fvm_test.go (hunk 2 above)
Comment (sideninja) [reply fvm/fvm_test.go:3019]: Right, thanks.

File: fvm/evm/types/events.go (hunk 3 above)
Comment (sideninja) [reply fvm/evm/types/events.go:78]: This is covered in the test I wrote, however edge cases with errors are not.

**Pull request merged**